packages/ketcher-react/src/script/editor/utils/selectItems/startSelecting.ts

```diff
@@ -0,0 +1,158 @@
+import { FunctionalGroup, SGroup } from 'ketcher-core'
+import { selMerge } from '../../tool/select'
+import { atomLongtapEvent } from '../../tool/atom'
+
+function isSelected(selection, item) {
+  return (
+    selection && selection[item.map] && selection[item.map].includes(item.id)
+  )
+}
+
+function closestToSel(closestItem) {
+  const res = {}
+  res[closestItem.map] = [closestItem.id]
+  return res
+}
+
+export function startSelecting(event, self, lassoHelper?) {
```

Comment:
code in onMouseDown in select.ts moved in startSelecting function with little or no updates